src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/LineLengthCheck.java

```diff
@@ -130,25 +171,44 @@
     /** Default maximum number of columns in a line. */
     private static final int DEFAULT_MAX_COLUMNS = 80;
 
-    /** Patterns matching package, import, and import static statements. */
-    private static final Pattern IGNORE_PATTERN = Pattern.compile("^(package|import) .*");
+    /** Pattern matching import and import static statements. */
+    private static final Pattern IMPORT_PATTERN = Pattern.compile("^import .*");
+
+    /** Pattern matching package statements. */
+    private static final Pattern PACKAGE_PATTERN = Pattern.compile("^package .*");
 
     /** Specify the maximum line length allowed. */
     private int max = DEFAULT_MAX_COLUMNS;
 
     /** Specify pattern for lines to ignore. */
     private Pattern ignorePattern = Pattern.compile("^$");
 
+    /** Validate checking of import statements. */
+    private boolean validateImport;
+
+    /** Validate checking of package statements. */
+    private boolean validatePackage;
+
     @Override
     protected void processFiltered(File file, FileText fileText) {
         for (int i = 0; i < fileText.size(); i++) {
             final String line = fileText.get(i);
             final int realLength = CommonUtil.lengthExpandedTabs(
                 line, line.codePointCount(0, line.length()), getTabWidth());
 
-            if (realLength > max && !IGNORE_PATTERN.matcher(line).find()
-                && !ignorePattern.matcher(line).find()) {
-                log(i + 1, MSG_KEY, max, realLength);
+            if (realLength > max && !ignorePattern.matcher(line).find()) {
+
+                final boolean matchPackage = PACKAGE_PATTERN.matcher(line).find();
+                final boolean matchImport = IMPORT_PATTERN.matcher(line).find();
+                final boolean validateForPackage = validatePackage && matchPackage;
+                final boolean validateForImport = validateImport && matchImport;
+
+                if (validateForImport || validateForPackage) {
+                    log(i + 1, MSG_KEY, max, realLength);
```

Comment:
The body of both `if` is the same as for `else if`. Suggested changes:
```java
if (realLength > max
     && isNotAllowedException(ignorePattern, true)
     && isNotAllowedException(IMPORT_PATTERN, validateImport)
     && isNotAllowedException(PACKAGE_PATTERN, validatePackage) {

  log(i + 1, MSG_KEY, max, realLength);
}
 
static boolean isNotAllowedException(Pattern allowed, boolean shouldValidate) {
 return shouldValidate &&!allowed.matcher(line).find();
}
```

Reply:
Added that test case. The above idea won't work (Just consider the case when `validateImport` is false, it will never log violations) I merged the if and elif statements.